packages/office-ui-fabric-react/src/utilities/grid/Grid.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { BaseComponent, getId } from '../../Utilities';
+import { IGridProps } from './Grid.Props';
+
+export class Grid extends BaseComponent<IGridProps, {}> {
+
+  // Used to track the index of item we are on while building grid
+  private _gridIndex;
+
+  private _id;
+
+  private _items: any[][];
+
+  constructor(props: IGridProps) {
+    super(props);
+    this._items = [];
+    this._id = getId();
+  }
+
+  public render() {
+    let {
+      items,
+      columnCount,
+      onRenderItem,
+      positionInSet,
+      setSize
+    } = this.props;
+
+    // Reset the items array so we will start fresh
+    this._items = [];
+
+    items.map((item, index) => {

```

Comment:
map returns an array, so it's usually used to convert some array to a different type of format, reduce is what you want to use to reduce the array to a single value (matrix):
For example (didn't test it):
function toMatrix<T>(array: T[], columns: number): T[][] {
  return array.reduce((acc, currentValue, currentIndex) => {
    if (currentIndex % columns === 0) {
      accumulator.push([currentValue]);
    } else {
      accumulator[accumulator.length - 1].push(currentValue);
    }
    return accumulator;
  }, [] as T[][]);
}